disperser/batcher/batch_confirmer.go

```diff
@@ -0,0 +1,531 @@
+		_ = b.handleFailure(ctx, blobs, FailNoAggregatedSignature)
+		return errors.New("failed to process confirmed batch: aggSig from transaction manager metadata is nil")
+	}
+	b.logger.Info("received ConfirmBatch transaction receipt", "blockNumber", receiptOrErr.Receipt.BlockNumber, "txnHash", receiptOrErr.Receipt.TxHash.Hex())
+
+	// Mark the blobs as complete
+	stageTimer := time.Now()
+	blobsToRetry, err := b.updateConfirmationInfo(ctx, confirmationMetadata, receiptOrErr.Receipt)
+	if err != nil {
+		_ = b.handleFailure(ctx, blobs, FailUpdateConfirmationInfo)
+		return fmt.Errorf("failed to update confirmation info: %w", err)
+	}
+	if len(blobsToRetry) > 0 {
+		b.logger.Error("failed to update confirmation info", "failed", len(blobsToRetry), "total", len(blobs))
+		_ = b.handleFailure(ctx, blobsToRetry, FailUpdateConfirmationInfo)
+	}
+	b.logger.Debug("Update confirmation info took", "duration", time.Since(stageTimer).String())
+	batchSize := int64(0)
+	for _, blobMeta := range blobs {
+		batchSize += int64(blobMeta.RequestMetadata.BlobSize)
+	}
+
+	return nil
+}
+
+func (b *BatchConfirmer) handleFailure(ctx context.Context, blobMetadatas []*disperser.BlobMetadata, reason FailReason) error {
+	var result *multierror.Error
+	numPermanentFailures := 0
+	for _, metadata := range blobMetadatas {
+		retry, err := b.BlobStore.HandleBlobFailure(ctx, metadata, b.MaxNumRetriesPerBlob)
+		if err != nil {
+			b.logger.Error("HandleSingleBatch: error handling blob failure", "err", err)
+			// Append the error
+			result = multierror.Append(result, err)
+		}
+
+		if retry {
+			continue
+		}
+
+		numPermanentFailures++
+	}
+
+	// Return the error(s)
+	return result.ErrorOrNil()
+}
+
+func (b *BatchConfirmer) HandleSingleBatch(ctx context.Context) error {
+	currentBlock, err := b.ethClient.BlockNumber(ctx)
+	if err != nil {
+		return fmt.Errorf("error getting current block number: %w", err)
+	}
+
+	err = b.EncodingStreamer.UpdateReferenceBlock(uint(currentBlock))
+	if err != nil {
+		return fmt.Errorf("error updating reference block number: %w", err)
+	}
+
+	// Get the pending batch
+	b.logger.Debug("Getting latest formed batch...")
+	stateUpdateTicker := time.NewTicker(b.DispersalStatusCheckInterval)
+	var batch *BatchRecord
+	var minibatches []*MinibatchRecord
+	for batch == nil {
+		select {
+		case <-ctx.Done():
+			return ctx.Err()
+		case <-stateUpdateTicker.C:
+			batch, minibatches, err = b.MinibatchStore.GetLatestFormedBatch(ctx)
+			if err != nil {
+				b.logger.Error("error getting latest formed batch", "err", err)
+			}
+		}
+	}
+
+	if len(minibatches) == 0 {
+		return fmt.Errorf("no minibatches in the batch %s", batch.ID)
+	}
+
+	// Make sure all minibatches in the batch have been dispersed
+	batchDispersed := false
+	stateUpdateTicker.Reset(b.DispersalStatusCheckInterval)
+	ctxWithTimeout, cancel := context.WithTimeout(ctx, b.DispersalTimeout)
+	defer cancel()
+	for !batchDispersed {
+		select {
+		case <-ctxWithTimeout.Done():
```

Comment:
What happens if this timeout is hit? What happens to the minibatches?

It seems possible that it will try to process this batch again the next round...

Reply:
Yes, that's the current behavior now. 
This timeout should be longer than the [dispersal timeout](https://github.com/Layr-Labs/eigenda/blob/master/disperser/batcher/minibatcher.go#L292), then all requests will have succeeded/failed before this timeout. 
In case batcher shutdown mid-dispersal, the minibatcher recovery logic should clear the dispersal requests with no matching responses. 
Is this too brittle?